stdlib/src/locale.rs

```diff
@@ -67,6 +67,44 @@ mod _locale {
         )
     }
 
+    #[pyfunction]
+    fn strcoll(string1: PyStrRef, string2: PyStrRef, vm: &VirtualMachine) -> PyResult {
+        unsafe {
+            let string1_cstr = CString::new(string1.as_str()).map_err(|e| e.to_pyexception(vm))?;
+            let string2_cstr = CString::new(string2.as_str()).map_err(|e| e.to_pyexception(vm))?;
+            Ok(vm.new_pyobj(libc::strcoll(string1_cstr.as_ptr(), string2_cstr.as_ptr())))
+        }
+    }
+
+    #[pyfunction]
+    fn strxfrm(string: PyStrRef, vm: &VirtualMachine) -> PyResult {
+        // https://github.com/python/cpython/blob/eaae563b6878aa050b4ad406b67728b6b066220e/Modules/_localemodule.c#L390-L442
+        unsafe {
+            let n1 = string.char_len() + 1;
+            let mut buff = vec![0; n1];
+            let buff_ptr = buff.as_mut_ptr();
+
+            let string_cstr = CString::new(string.as_str()).map_err(|e| e.to_pyexception(vm))?;
+            let string_ptr = CStr::as_ptr(&string_cstr);
+            let n2 = libc::strxfrm(buff_ptr, string_ptr, n1) + 1;
+
+            if n2 >= n1 {
+                let mut new_buff = vec![0; n2];
+                // this part corresponds to re-allocate in CPython
+                for (i, item) in buff.iter().enumerate() {
+                    if *item != 0 {
+                        new_buff[i] = *item;
+                    }
+                }
+
+                let new_buff_ptr = new_buff.as_mut_ptr();
+                libc::strxfrm(new_buff_ptr, string_ptr, n2);
```

Comment:
A trick to reuse the given length.
```suggestion
                let n2 = libc::strxfrm(buff.as_mut_ptr(), string_ptr, n2);
                buff.trancate(n2);
                vm.new_pyobj(String::from_utf8(buff).expect("strxfrm returned invalid utf-8 string"))
```